### DISCUSSION TITLE
Adding repeatable comment to all instances of a symbol

### DISCUSSION THREAD

**Discussion by TimRegeant** [NationalSecurityAgency/ghidra#6225] [Q&A] [answered]
How can I have the listing automatically show a repeatable comment for all instances of a symbol?

For example, in the code below I would like to have an predefined EOL comment automatically entered for all instances of the symbol hw_MFP_IERA.

      000180dc 08 f8 00      bset.b    0x0,(hw_MFP_IERA).w
               00 fa 07
      000180e2 08 f8 00      bset.b    0x0,(hw_MFP_IMRA).w
               00 fa 13
      000180e8 4e 75         rts
      000180ea 08 b8 00      bclr.b    0x0,(hw_MFP_IERA).w
               00 fa 07

**Comment by ryanmkurtz** [comment]
Repeatable comments work with cross references.  You put it on the source of your data, and any cross reference location will also show the comment.

**Reply by TimRegeant** [reply]
Thanks, could you elaborate on how to set cross references?  When you say "You put it on the source of your data" what does that mean?

**Reply by ryanmkurtz** [reply]
If you double click on `hw_MFP_IERA`, does it take you somewhere?  If so, put your repeatable comment at that location.

**Comment by TimRegeant** [comment]
No, it doesn't take me anywhere.  It is an external hardware address for the platform the code is to be run on.

**Reply by ryanmkurtz** [reply]
So is `hw_MFP_IERA` in the Symbol Table?  Or is it an Equate?  Do you know where it came from?

**Comment by TimRegeant** [comment]
Yes, I have it in the symbol table.

**Reply by ryanmkurtz** [reply]
Does it have an address?

**Reply by TimRegeant** [reply]
Yes, it it 0xFFFA07.

**Reply by ryanmkurtz** [reply]
I might be able to help better if you can attach your program.  `File -> Export Program -> Ghidra Zip File`.

**Reply by TimRegeant** [reply]
Thanks for the help. Here is a code segment exported as Ghidra Zip File (of course the .txt extension is just there to allow the upload).

It has symbols loaded and 1 repeatable comment already set for symbol hw_MFP_IERA (0xfffffa07).

There are two references in the stub to hw_MFP_IERA so hopefully should be good to test the repeatable comment functionality.

[memdump-stub.gzf.txt](https://github.com/NationalSecurityAgency/ghidra/files/14269677/memdump-stub.gzf.txt)

**Reply by ryanmkurtz** [reply, marked as answer]
The symbol address are not in memory.  You need to make a new memory block that they live in.  See how all the symbols are red in the symbol table?  Once you make the memory block, they should not longer be red and everything should hopefully work (it did for me).

**Reply by TimRegeant** [reply]
Thanks so much!
